### PR TITLE
feat(admin): dashboard with KPI cards, student table, and analytics (#383)

### PR DESCRIPTION
**Pull request by okpilot**
## Summary

- Add admin dashboard route at `/app/admin/dashboard` with nav entry, types, searchParams parsing, and loading skeleton
- Create 3 SECURITY DEFINER RPCs: `get_admin_dashboard_kpis`, `get_admin_weak_topics`, `get_admin_student_stats` with auth + admin + org-scoping
- Add query functions calling RPCs + Supabase joins, with TS-side merge/sort/paginate for student table
- Add DashboardHeader client component with time range Select (7d/30d/90d/all)
- 61 unit tests (parseFilters + query functions)
- Document all RPCs in database.md

**Note:** Tasks 5-10 (UI components, page composition, student detail page) are in progress. This PR is pushed early for CodeRabbit to start reviewing the data layer while UI work continues.

**Known gap:** `users.last_active_at` is never populated — tracked in #479.

## Test plan

- [x] `pnpm check-types` passes (0 errors)
- [x] `pnpm test` passes (177 files, 2394 tests)
- [x] Security auditor: 0 CRITICAL/HIGH
- [x] parseFilters tests: all enum guards, defaults, edge cases (33 tests)
- [x] Query function tests: merge, sort, paginate, error paths (28 tests)
- [ ] Manual: navigate to `/app/admin/dashboard` as admin, verify skeleton loads
- [ ] E2E: admin dashboard spec (Task 10)

🤖 Generated with [Claude Code](https://claude.com/claude-code)

## Summary by CodeRabbit

* **New Features**
  * Full Admin Dashboard: KPI overview, time-range selector, recent activity, weakest topics, interactive student table with filtering/sorting/pagination, student detail view, session history and session report drill‑downs, plus loading skeletons and empty states.

* **Documentation**
  * Updated database RPC/docs with admin analytics/reporting behavior and last-active stamping.

* **Tests**
  * Extensive tests across dashboard UI, tables, reports, and utilities.

* **Chores**
  * DB migrations, seed script, type regen, and dev tooling updates.

### CODE REVIEW

File: apps/web/app/app/admin/dashboard/queries.ts [hunk 1]
```diff
@@ -0,0 +1,224 @@
...
+    .eq('organization_id', organizationId)
+    .eq('role', 'student')
+
+  // Default: show active students. RPC only returns stats for non-deleted students,
+  // so deleted students would show zeroed metrics — filter them out unless explicitly requested.
+  if (filters.status === 'inactive') {
+    query = query.not('deleted_at', 'is', null)
+  } else {
+    query = query.is('deleted_at', null)
+  }
```
[the remark this answers is not in the document]
Comment (okpilot) [reply apps/web/app/app/admin/dashboard/queries.ts]: Already fixed in 0cf8b9f. When `status` is undefined ("All Students"), neither `deleted_at` filter is applied — the query returns both active and inactive students. See lines 84-90 with inline comment documenting the behavior.

**Comment by okpilot** [comment]
## Addressing outside-diff review comments

**packages/db/src/types.ts (lines 1062-1168)** — Missing RPC type signatures for `get_admin_dashboard_kpis`, `get_admin_weak_topics`, `get_admin_student_stats`.

Accurate finding. The migrations exist but types haven't been regenerated against a DB with them applied. Currently worked around with a `RpcFn` type cast in `queries.ts` (lines 17-24) with an explicit comment noting the gap.

Deferred to #490 — requires running `supabase gen types typescript` against a database with all migrations applied. Not blocking for correctness (all RPC calls work at runtime).

File: apps/web/app/app/admin/dashboard/_components/student-table-helpers.test.tsx [hunk 2]
```diff
@@ -0,0 +1,380 @@
...
+    render(
+      <table>
+        <tbody>
+          <StudentRow student={buildStudent({ isActive: false })} onClick={onClick} />
+        </tbody>
+      </table>,
+    )
+    fireEvent.keyDown(screen.getByRole('row'), { key: ' ' })
+    expect(onClick).not.toHaveBeenCalled()
+  })
```
[the remark this answers is not in the document]
Comment (okpilot) [reply apps/web/app/app/admin/dashboard/_components/student-table-helpers.test.tsx]: Fixed in 56bf077. Added `Array.isArray(usersData)` guard at line 111 for consistency with the three sibling sites. Also added 3 new tests covering the `Array.isArray` else-branch (non-array RPC responses) for `getDashboardStudents`, `getWeakTopics`, and `getRecentSessions`.

[the remark this answers is not in the document]
Comment (okpilot) [reply apps/web/app/app/admin/dashboard/_components/student-table-helpers.test.tsx]: Apologies for the crossed wires in the previous reply. The inactive mouse-click test was added in e35e090 at line 359: renders StudentRow with `isActive: false`, fires `fireEvent.click`, asserts `onClick` not called. This mirrors the existing Enter/Space inactive tests at lines 333-357.

File: apps/web/app/app/admin/dashboard/queries.ts [hunk 3]
```diff
@@ -0,0 +1,230 @@
...
+      case 'mastery':
+        return dir * (a.mastery - b.mastery)
+      default:
+        return 0
+    }
+  })
+
+  const totalCount = merged.length
+  const start = (filters.page - 1) * PAGE_SIZE
+  const students = merged.slice(start, start + PAGE_SIZE)
```
[the remark this answers is not in the document]
Comment (okpilot) [reply apps/web/app/app/admin/dashboard/queries.ts]: False positive — every sort branch already includes an `|| a.id.localeCompare(b.id)` tie-breaker (lines 136, 140, 144, 146, 148). Rows with identical sort keys are deterministically ordered by UUID, so pagination is stable across requests.

File: apps/web/lib/queries/admin-quiz-report.ts [hunk 4]
```diff
@@ -0,0 +1,198 @@
...
+  if (sessionError) {
+    console.error('[getAdminQuizReportSummary] Session query error:', sessionError.message)
+    return null
+  }
+  const session = sessionData as AdminSessionRow | null
+  if (!session) return null
+  // Only serve reports for completed sessions — prevents mid-session answer exposure
+  if (!session.ended_at) return null
+
+  // Session org-membership verified on lines 50-58 above — sessionId is safe to use unscoped
```
[the remark this answers is not in the document]
Comment (okpilot) [reply apps/web/lib/queries/admin-quiz-report.ts]: Fixed in c346042. Removed the incorrect line numbers — comment now reads "Session org-membership verified above".
